examples/codeviewer/common/src/jvmMain/kotlin/org/jetbrains/codeviewer/platform/JvmFile.kt

```diff
@@ -87,15 +89,15 @@ private fun java.io.File.readLinePositions(
             val ib = channel.map(
                 FileChannel.MapMode.READ_ONLY, 0, channel.size()
             )
-            var isBeginOfLine = true
             var position = 0L
+            starts.add(0)
             while (ib.hasRemaining()) {
                 val byte = ib.get()
+                position++
+                val isBeginOfLine = byte.toInt().toChar() == '\n'
```

Comment:
If I am not mistaken, in the previous version `\n` was in the end of a line. Now it is in the begining of a line.

By design, we should create a new line after `\n`, not before. Not sure why it works in codeviewer. Maybe because of `maxLines = 1`.

If it works, okay, but it is better to make a fix differently, so people won't be confused.

For example, we can remove these characters here, not in `get(index: Int)`. Or check for 2 symbols in `get(index: Int)`

Reply:
No, this is not changed. This code finds indexes of `\n`, and previously the problem was that it didn't add the last one.

Your confusion is probably because of the fact that previous implementation adds the position **only on next loop iteration**.

Let me try to refactor it/write comments to make it more readable

Reply:
Indeed. The results are not changed:

Before:
![image](https://user-images.githubusercontent.com/5963351/226876687-5d7badbf-35bf-4df6-9cb7-b9555af43b42.png)

After:
![image](https://user-images.githubusercontent.com/5963351/226877127-2f62bd53-67c8-45e3-ac72-92dda399af4b.png)

The confusion because of `byte` corresponding to `position-1` now (not `position`). We can rename it to `previousByte`